libsplinter/src/circuit/template/mod.rs

```diff
@@ -364,6 +427,40 @@ rules:
             .any(|(key, value)| key == "peer_services" && value == "[\"a000\"]"));
     }
 
+    /// Verifies a `CircuitTemplateManager` can be created and a template can be loaded from the
+    /// `SPLINTER_CIRCUIT_TEMPLATE_PATH` environment variable
+    ///
+    /// The test follows the procedure below:
+    /// 1. Sets up a temporary directory, to write a circuit template YAML file from the
+    ///    `EXAMPLE_TEMPLATE_YAML`.
+    /// 2. Sets the `SPLINTER_CIRCUIT_TEMPLATE_PATH` environment variable based on the temporary
+    ///    directory.
+    /// 3. Create a `CircuitTemplateManager` using the environment variable, assert this is
+    ///    successfully constructed.
+    /// 4. Load a circuit template using the `CircuitTemplateManager` and assert this is
+    ///    successfully loaded.
+    /// 5. Load a circuit template using the `CircuitCreateTemplate` `from_yaml_file` method and
+    ///    assert this is successfully loaded.
+    ///
+    /// Verifying the last 3 steps were done successfully verifies the `SPLINTER_CIRCUIT_TEMPLATE_PATH`
+    /// environment variable has been successfully set and used by the `CircuitTemplateManager` and
+    /// `CircuitCreateTemplate` modules to load a circuit template yaml file.
+    #[test]
+    fn test_template_from_env() {
+        let temp_dir = TempDir::new("test_builds_template_v1").unwrap();
+        let temp_dir = temp_dir.path().to_path_buf();
+        std::env::set_var("SPLINTER_CIRCUIT_TEMPLATE_PATH", &temp_dir);
```

Comment:
This test should take care to reset SPLINTER_CIRCUIT_TEMPLATE_PATH to its original value (maybe making this a hard function to write properly). Obviously this would conflict with any other test using SPLINTER_CIRCUIT_TEMPLATE_PATH.